app/code/core/Mage/Adminhtml/Block/Customer/Online/Grid/Renderer/Url.php

```diff
@@ -35,6 +35,6 @@ class Mage_Adminhtml_Block_Customer_Online_Grid_Renderer_Url extends Mage_Adminh
      */
     public function render(Varien_Object $row)
     {
-        return htmlspecialchars($row->getData($this->getColumn()->getIndex()));
+        return htmlspecialchars((string)$row->getData($this->getColumn()->getIndex()));
```

Comment:
No need for `htmlspecialchars()` if empty value:

```suggestion
       $value = $row->getData($this->getColumn()->getIndex());
       return empty($value) ? '' : htmlspecialchars($value);
```

Reply:
Instead of `empty()` check `is_string()` might be more clear.

Working on it ...